posts/2025-03-20-igraph-isc-setting-success-decade/index.qmd

```diff
@@ -0,0 +1,90 @@
+Over the years, the R package has also accrued a bit of documentation debt and technical debt, that we wanted to tackle to set up igraph for success in the next decade.
+Both aspects had the potential to simplify further updates of the package.
+
+## Adopting the lifecycle system
+
+To provide a more consistent interface, part of igraph functions or arguments are slowly but consistently being deprecated.
+Some of them used to be deprecated using custom solutions or base R deprecations.
+In contrast, now all deprecations happen through the lifecycle package.
+On manual pages, deprecation badges indicate the maintenance status of functions or arguments.
+
+```r
+library("igraph")
+#> 
+#> Attaching package: 'igraph'
+#> The following objects are masked from 'package:stats':
+#> 
+#>     decompose, spectrum
+#> The following object is masked from 'package:base':
+#> 
+#>     union
+g <- sample_gnp(10, 0.5)
+graph.density(g)
+#> Warning: `graph.density()` was deprecated in igraph 2.0.0.
+#> ℹ Please use `edge_density()` instead.
+#> This warning is displayed once every 8 hours.
+#> Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
+#> generated.
+#> [1] 0.5111111
+```
+
+Having everything in the "lifecycle system" now means the interface and documentation is consistent for users, and also that we can more easily increase deprecation levels over time by searching for lifecycle calls.
+Last, but not least, all deprecations are listed in a [vignette](https://r.igraph.org/articles/current-deprecations.html).
+
+## Documentation improvements
+
+Besides the documentation of lifecycle, we created and used a custom roxygen2 tags to link to the C docs from manual pages.
+[Example](https://r.igraph.org/reference/sample_sbm.html#related-documentation-in-the-c-library).
+
+## Fewer errors, better errors
+
+We have made a conscious effort to port errors from base R to cli, at the same time improving phrasing of errors.
+We have also fixed some bugs ( [example](https://github.com/igraph/rigraph/pull/1716)).
+
+## Work on the internals
+
+We did a lot of refactoring!
+Most of it was focussed on [tests](https://github.com/igraph/rigraph/pulls?q=sort%3Aupdated-desc+is%3Apr+in%3Atitle+%22test%3A%22+is%3Amerged), as [previously explained on this blog](/posts/2025-03-04-refactoring-test-files/): merging some test files to ensure alignment between R scripts and test files, refactoring test files for better readability and future debugging, updating the testthat usage by removing old expectations.
+We also worked in the `R/`folder, for instance we replaced the usage of an embedded version of lazyeval with rlang calls.
+( [PR 1](https://github.com/igraph/rigraph/pull/1441), [PR 2](https://github.com/igraph/rigraph/pull/1445))
+
+## Sharing our experience
+
+We shared our experience working on igraph in part of a talk and in various blog posts:
+
```

Comment:
Thanks for this external reference list. Could you add a brief statement of the most outstanding learnings in this blog post?

Reply:
I'd prefer not to as it's a summary post.